lib/pbench/test/unit/server/test_datasets_metadata.py

```diff
@@ -0,0 +1,178 @@
+from http import HTTPStatus
+import itertools
+
+from pbench.server.api.resources import ParamType
+from pbench.server.api.resources.datasets_metadata import DatasetsMetadata
+
+
+class TestDatasetsMetadata:
+    def test_get_no_dataset(self, client, server_config, attach_dataset):
+        response = client.get(
+            f"{server_config.rest_uri}/datasets/metadata",
+            query_string={
+                "controller": "node",
+                "name": "foobar",
+                "metadata": ["seen", "saved"],
+            },
+        )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+        assert response.json == {"message": "Dataset foobar not found"}
+
+    def test_get_bad_keys(self, client, server_config, attach_dataset):
+        response = client.get(
+            f"{server_config.rest_uri}/datasets/metadata",
+            query_string={
+                "controller": "node",
+                "name": "foobar",
+                "metadata": ["xyzzy", "plugh", "owner", "access"],
+            },
+        )
+        assert response.status_code == HTTPStatus.BAD_REQUEST
+        assert response.json == {"message": "Dataset foobar not found"}
```

Comment:
Looking at line 31, I donna thin this test does awhat it says it does.  (I.e., shouldn't the error message be complaining about those adventurous keys not being found?). [Was that game _Adventure_ or _Zork_??]

Reply:
It looks like you fixed this for the `PUT` test.

Reply:
Too much copy-and-paste! Oops. "Works as coded"... too bad it doesn't actually test anything interesting.